### PR TITLE
PROCESS : Integrate eco-design tool to CI

### PR DESCRIPTION
**Pull request by vallhallalm**
I, Louis Magnier, hereby grant to Hyperglosae maintainers the right to publish our contribution under the terms of any licenses the Free Software Foundation classifies as Free Software Licenses.

Just a little more details and things to do so that this process improvement integrate themselves nicely in the process :

- As we do commit some file within a workflow we will need to grant them the right to write in our repository.
- Greenframe need a new secret to be set up to work perfectly. This secret has to be named GREENFRAME_SECRET_TOKEN and needs to contain the token that greenframe provide
- Before executing the new workflow it might be necessary to create a new orphan branch called htmlPreview with the following command
```sh
git checkout --orphan htmlPreview
git rm -rf .
rm '.gitignore'
echo "# This branch is meant to preview html output of Green It Analysis analyse" > README.md
git add README.md
git commit -a -m "Initial Commit"
git push origin htmlPreview
``
- Regarding the analysis conducted with greenframe, there is a threshold which needs to be set manually. For example if we know that we are going to commit a change which might be quite energy consuming but that is necessary, it might be interesting to raise the threshold before the commit so that tests do not fail. Threshold can be set in /tests/greenframeThreshold.sh on line 5

### CODE REVIEW

**Comment by benel** [comment]
Thank you very much @vallhallalm. I followed your instructions to set up the repository, especially with a repository secret.

It seems however that: 

> Anyone with collaborator access to this repository can use these **secrets** and variables for actions. They are **not passed** to workflows that are **triggered by a pull request** from a fork.

In other words, integrating the new "action" would result in that any future pull requests would fail.

I see two different workarounds. Could you please think about their pros and cons? 

We will discuss this next time and choose one of them.

File: tests/greenIT/input/url.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Analyse l'URL collectif.greenit.fr 
+- name : 'HyperGlosae'
+  url : 'http://localhost:3000/'
```
Comment (benel) [inline tests/greenIT/input/url.yaml:3]: @vallhallalm Notez que la page d'accueil est la seule page qui n'a pas encore réellement fait l'objet d'un design et encore moins d'un "eco-design". Je comprends l'intérêt d'analyser sa qualité pour illustrer une amélioration, mais par contre ce serait un peu trompeur d'analyser uniquement celle-là.

Comment (vallhallalm) [reply tests/greenIT/input/url.yaml:3]: @benel Effectivement je tacherai de complexifier le scenario de green IT analysis quand j'aurais réussi à faire passer le test sur la pull request

**Comment by vallhallalm** [comment]
@benel I managed to delete the visit of the index page in the scenario for greenframe integration (as we discussed this morning) and it cut about half of the consumption estimation so now the test is passing successfully

**Comment by benel** [comment]
Done with bffef5443ecb5731e99d110ca5f1150c90bb8d41, a6dc1cd9bf94ceac3c28f5c055bcd3c0f408a1db and c8629e3e6219ecbdb1b4a11074f714439beec6f2.

**Pull request closed without merging**